File: aat-lib/src/main/java/ch/bailu/aat_lib/map/tile/source/CachedSource.java
package ch.bailu.aat_lib.map.tile.source;

import org.mapsforge.core.model.Tile;

import ch.bailu.aat_lib.app.AppContext;
import ch.bailu.aat_lib.service.cache.Obj;
import ch.bailu.aat_lib.service.cache.ObjTileCached;

public class CachedSource extends Source {
    private final Source source;

    public CachedSource(Source s) {
        source = s;
    }

    public String getName() {
        return "Cached" + source.getName();
    }

    @Override
    public String getID(Tile aTile, AppContext context) {
        return "Cached" + source.getID(aTile, context);
    }

    @Override
    public int getMinimumZoomLevel() {
        return source.getMinimumZoomLevel();
    }

    @Override
    public int getMaximumZoomLevel() {
        return source.getMaximumZoomLevel();
    }

    @Override
    public int getAlpha() {
        return source.getAlpha();
    }


    @Override
    public Obj.Factory getFactory(Tile tile) {
        return new ObjTileCached.Factory(tile, source);
    }

    @Override
    public boolean isTransparent() {
        return source.isTransparent();
    }


    public final static CachedSource CACHED_ELEVATION_HILLSHADE = new CachedSource(ElevationSource.ELEVATION_HILLSHADE);
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/map/tile/source/CachedSource.kt
package ch.bailu.aat_lib.map.tile.source

import ch.bailu.aat_lib.app.AppContext
import ch.bailu.aat_lib.service.cache.Obj
import ch.bailu.aat_lib.service.cache.ObjTileCached
import org.mapsforge.core.model.Tile

class CachedSource(private val source: Source) : Source() {
    override val name: String
        get() = "Cached" + source.name

    override fun getID(tile: Tile, context: AppContext): String {
        return "Cached" + source.getID(tile, context)
    }

    override val minimumZoomLevel: Int
        get() = source.minimumZoomLevel

    override val maximumZoomLevel: Int
        get() = source.maximumZoomLevel

    override val alpha: Int
        get() = source.alpha


    override fun getFactory(tile: Tile): Obj.Factory {
        return ObjTileCached.Factory(tile, source)
    }

    override val isTransparent: Boolean
        get() = source.isTransparent


    companion object {
        val CACHED_ELEVATION_HILLSHADE: CachedSource =
            CachedSource(ElevationSource.ELEVATION_HILLSHADE)
    }
}
